edit-post/components/header/fixed-toolbar-toggle/index.js

```diff
@@ -1,49 +1,41 @@
 /**
- * External Dependencies
+ * WordPress Dependencies
  */
-import { connect } from 'react-redux';
+import { withSelect, withDispatch } from '@wordpress/data';
 
 /**
  * WordPress Dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { compose } from '@wordpress/element';
 import { MenuItemsGroup, MenuItemsToggle, withInstanceId } from '@wordpress/components';
+import { ifViewportMatches } from '@wordpress/viewport';
 
-/**
- * Internal Dependencies
- */
-import { hasFixedToolbar, isMobile } from '../../../store/selectors';
-import { toggleFeature } from '../../../store/actions';
-
-function FeatureToggle( { onToggle, active, onMobile } ) {
-	if ( onMobile ) {
-		return null;
-	}
+function FeatureToggle( { onToggle, isActive } ) {
 	return (
 		<MenuItemsGroup
 			label={ __( 'Settings' ) }
 			filterName="editPost.MoreMenu.settings"
 		>
 			<MenuItemsToggle
 				label={ __( 'Fix Toolbar to Top' ) }
-				isSelected={ active }
+				isSelected={ isActive }
 				onClick={ onToggle }
 			/>
 		</MenuItemsGroup>
 	);
 }
 
-export default connect(
-	( state ) => ( {
-		active: hasFixedToolbar( state ),
-		onMobile: isMobile( state ),
-	} ),
-	( dispatch, ownProps ) => ( {
+export default compose( [
```

Comment:
See how `compose` is implemented in Ramda: http://ramdajs.com/docs/#compose. Lodash allows to pass an array, but in the FP world it is a function that takes any number of functions:
> ((y → z), (x → y), …, (o → p), ((a, b, …, n) → o)) → ((a, b, …, n) → z)

Reply:
>Lodash allows to pass an array, but in the FP world it is a function that takes any number of functions:

Yeah, I don't feel strongly one way or the other, but knowing how `arguments` works with [`Function#length`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/length) and [various V8 deoptimizations](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments), I am (apathetically) cautious to avoid it for fear of encouraging its use.

Tangentially related: See [`classcat`](https://github.com/jorgebucaran/classcat) vs. [`classnames`](https://github.com/JedWatson/classnames) where the difference is passing as array vs. arguments respectively, with the former benefitting as such both in performance and bundle size.

Reply:
Feel free to measure the impact in here and add Eslint rule so we could fly with one style 👍
Speaking myself, it’s also tempting to teach Babel to optimize our code, in cases like this. I’m fine with the surrounding array syntax but I’d love to see one style everywhere.